### ISSUE TITLE
Replace all hyphens (-) in selection with colons (:)

### ISSUE THREAD

**Issue by i1own0u** [Pitmairen/selection-search#76]
I know I'm not great with regex, but I can't figure out how to replace hyphens with colons.  I can replace it with "+"

{%s|re:\-+:::+:::g} ##changes all dashes to plus (+)
0-0 ---> 0+0

{%s|re:\-+:::::::g} ##no change
0-0 ---> 0-0

{%s|re:\-+:::\::::g} ##no change
0-0 ---> 0-0

{%s|re:\-+:::\:\:::g} ##changes all dashes to \:
0-0 ---> 0\:0

What expression am I supposed to be doing to get the following result?
0-0 ---> 0:0

**Comment by Pitmairen** [comment]
Its a bug, I don't think its possible until the bug is fixed. The extra colon messes up the parsing of the regex. 

I should probably also try to improve the regex syntax, the triple colon syntax is not very nice. 

I think the only thing that can be done until it is fixed is to add a space after the colon, but then you have an extra space.

**Comment by i1own0u** [comment]
Sounds good to me!  Thanks.

**Comment by Pitmairen** [comment]
Yes, I was just thinking about that, as you can use multiple re expressions. Something like this should work:
```
{%s|re:-+:::\:\:::g|re:\\::::::g} 
```
This replaces the - with \:\ and then the second re, removes the  added \\.  Your versions probably also works.


I think I will reimplement the regex replacement using something like this:
```
{%s|replace:/-+/:/g} 
```
There the '/' separator is dynamic. The first letter in the expression will be the separator, so this will also work:
```
{%s|replace:#-+#:#g} 
```
Then it should be many separators to chose from in case someone want to use some of the separator characters in the expression.

**Comment by i1own0u** [comment]
Approach seems better -- can't see issues with it.

**Comment by Pitmairen** [comment]
The improved "replace:" converter should be available in version 0.8.53.
